### ISSUE TITLE
SNCosmo 2.0 NumPy requirement is out of date - Import Error

### ISSUE THREAD

**Issue by djperrefort** [sncosmo/sncosmo#245]
SNCosmo only requires `numpy >= 1.7.0`. However, when running NumPy version 1.15.4, importing SNCosmo 2.0 fails with the following error.

```
>>> import sncosmo
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/python3.7/site-packages/sncosmo/__init__.py", line 133, in <module>
    from .models import *
  File "/python3.7/site-packages/sncosmo/models.py", line 25, in <module>
    from .salt2utils import BicubicInterpolator, SALT2ColorLaw
  File "__init__.pxd", line 918, in init sncosmo.salt2utils
ValueError: numpy.ufunc size changed, may indicate binary incompatibility. Expected 216 from C header, got 192 from PyObject

```

This error is not raised for the most recent version of Numpy 1.16.4.

**Comment by kbarbary** [comment]
Did you install with conda? This might be an issue with how conda builds are specific to numpy runtime versions.

**Comment by djperrefort** [comment]
No - I installed all packages using pip.

**Comment by kbarbary** [comment]
Thanks that's helpful to know. I believe, but am not certain, that this is due to the new use of `pyproject.toml` to install build-time dependencies. Basically, at build time, pip is installing numpy and cython into an isolated environment, regardless of whether they are already installed on your system. It uses the most recent versions by default (numpy 1.16). Then sncosmo.salt2utils is build against that numpy (e.g., 1.16). But then at runtime, you might be using an older numpy, and get a failure.

I think the solution will be to  fix the version of numpy used in `pyproject.toml` to the oldest supported version. But I have to do some reading to check that this is really the best practices approach, and how old a version to use.

(I believe sncosmo is in fact compatible with numpy >= 1.7 in terms of the API. The binary compatibility is a different issue, but an important one!)

**Comment by kbarbary** [comment]
Similar issue: astrofrog/fast-histogram#35

**Comment by kbarbary** [comment]
Should be fixed in v2.1.0

**Comment by benjaminrose** [comment]
@djperrefort, do you still have this error or should we close this issue as fixed?

**Comment by kboone** [comment]
This should be permanently fixed with #305. We now rely on the scipy oldest-supported-numpy package to make sure that we are using compatible numpy versions for builds.
